hibernate-envers/src/test/java/org/hibernate/envers/test/integration/basic/BasicTypeColumnDefinitionTest.java

```diff
@@ -0,0 +1,156 @@
+ * its schema based on the right type rather than directly using the column definition as-is.
+ *
+ * The following illustrate some examples of expected transformations:
+ *
+ * <li>{@code @Column(columnDefinition = "varchar(10) not null")} => {@code sql-type = "varchar(255)"}</li>
+ * <li>{@code @Column(length = 10, columnDefinition = "varchar(10) not null")} => {@code sql-type = "varchar(10)"}</li>
+ * <li>{@code @Column(columnDefinition = "integer not null auto_increment")} => {@code sql-type = "integer"}</li>
+ *
+ * It is important to point out that resolving the sql-types length/precision/scale is all based on the
+ * values supplied as part of the {@link Column} annotation itself and not what is in the definition text.
+ *
+ * @author Chris Cranford
+ */
+@TestForIssue(jiraKey = "HHH-10844")
+@RequiresDialect(value = H2Dialect.class)
+public class BasicTypeColumnDefinitionTest extends BaseEnversJPAFunctionalTestCase {
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { BasicTypeContainer.class };
+	}
+
+	@Test
+	@Priority(10)
+	public void testMetadataBindings() {
+		final Table auditTable = metadata().getEntityBinding( BasicTypeContainer.class.getName() + "_AUD" ).getTable();
+
+		final org.hibernate.mapping.Column caseNumber = auditTable.getColumn( toIdentifier( "caseNumber" ) );
+		assertEquals( "integer", caseNumber.getSqlType() );
+		assertEquals( DEFAULT_LENGTH, caseNumber.getLength() );
+		assertEquals( DEFAULT_PRECISION, caseNumber.getPrecision() );
+		assertEquals( DEFAULT_SCALE, caseNumber.getScale() );
+
+		final org.hibernate.mapping.Column colDef = auditTable.getColumn( toIdentifier( "columnWithDefinition" ) );
+		assertEquals( "varchar(10)", colDef.getSqlType() );
+		assertEquals( 10, colDef.getLength() );
+		assertEquals( DEFAULT_PRECISION, colDef.getPrecision() );
+		assertEquals( DEFAULT_SCALE, colDef.getScale() );
+	}
+
+	@Test
+	@Priority(10)
+	public void initData() {
+		final BasicTypeContainer detachedEntity = doInJPA( this::entityManagerFactory, entityManager -> {
+			final BasicTypeContainer entity = new BasicTypeContainer();
+			entity.setData( "test" );
+			entity.setColumnWithDefinition( "1234567890" );
+			entityManager.persist( entity );
+			return entity;
+		} );
+
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			final BasicTypeContainer entity = entityManager.find( BasicTypeContainer.class, detachedEntity.getId() );
+			entity.setData( "test2" );
+			entityManager.merge( entity );
+		} );
+	}
+
+	@Test
+	public void testRevisionHistory() {
+		assertEquals( 2, getAuditReader().getRevisions( BasicTypeContainer.class, 1 ).size() );
+
+		final BasicTypeContainer rev1 = getAuditReader().find( BasicTypeContainer.class, 1, 1 );
+		assertEquals( "test", rev1.getData() );
+		assertEquals( "1234567890", rev1.getColumnWithDefinition() );
+		assertEquals( Integer.valueOf( 1 ), rev1.getCaseNumber() );
```

Comment:
could we rely on autoboxing above without explicitly creating a boxing type?

Reply:
Doing so would lead to an ambiguous call to `assertEquals` as the last argument is `Integer` and the former would be an `int.

Reply:
thanks for letting me know. Yeah, it does make sense.